src/compiler/Stage/InferTypes.elm

```diff
@@ -251,6 +251,19 @@ generateEquations ( expr, type_ ) =
                 :: generateEquations body
                 ++ bindingEquations
 
+        Typed.List list ->
+            case type_ of
+                Type.List type1 ->
+                    List.foldl
+                        (\(entry, type2) acc ->
+                            (equals type1 type2) :: acc
```

Comment:
I'd just change the names a bit:
* `type1` is `listParamType`
* `entry` we don't care about :arrow_right: `_`
* `type2` can be `listElementType`
* and `acc` is `equations` or `eqs`?

Anyway that's how I'd write it. :sweat_smile: